### ISSUE TITLE
🐛 [firebase_auth] Firebase Auth Web loses persistence after upgrading to the latest version

### ISSUE THREAD

**Issue by jomiplaz** [firebase/flutterfire#9089]
## Bug report

**Describe the bug**
Firebase Auth Web loses persistence after upgrading to the latest version.

**Steps to reproduce**
1. Login
2. Refresh the page

**Comment by sgehrman** [comment]
This is a show-stopper.  I can't release a new build until this is fixed.

$ flutter doctor -v
[✓] Flutter (Channel stable, 3.0.5, on Debian GNU/Linux 11 (bullseye) 5.10.0-16-amd64, locale en_US.UTF-8)
    • Flutter version 3.0.5 at /home/steve/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (31 hours ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0-rc4)
    • Android SDK at /home/steve/Android/Sdk
    • Platform android-31, build-tools 31.0.0-rc4
    • Java binary at: /home/steve/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • Debian clang version 11.0.1-2
    • cmake version 3.15.2
    • ninja version 1.8.2
    • pkg-config version 0.29.2

[✓] Android Studio (version 2021.2)
    • Android Studio at /home/steve/android-studio
    • Flutter plugin version 68.1.2
    • Dart plugin version 212.5744
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.69.1)
    • VS Code at /usr/share/code
    • Flutter extension version 3.44.0

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Debian GNU/Linux 11 (bullseye) 5.10.0-16-amd64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 103.0.5060.114

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

**Comment by sgehrman** [comment]
Nothing fancy in my code.  I listen to the authStateChanges stream.  I get null user on browser refresh.  I can login, I see the user, then refresh and I get back null for user.

**Comment by fanchungchit** [comment]
Same issue found here.
I've also tried to downgrade the package.
Before:
firebase_core: ^1.19.2
firebase_auth: ^3.4.2
Now:
firebase_core: 1.19.0
firebase_auth: 3.4.0

The result is getting this error.
TypeError: Cannot read properties of undefined (reading 'auth')
Which only happen on web.

**Comment by cspyo** [comment]
I solved a issue similar to this, following steps.

1. Delete your `pubspec.lock` file

2. Edit `pubspec.yaml` to downgrade the packages
```
firebase_auth: 3.4.1
firebase_auth_platform_interface: 6.3.1
firebase_core: 1.19.1
firebase_core_web: 1.6.6
```
  
3. Add this line in your `main`
```
if (kIsWeb) {
    await FirebaseAuth.instance.authStateChanges().first;
  }
```

You can get auth also after pressing a refresh button on Web.

**Comment by sgehrman** [comment]
I'm waiting for the update.  Is this a high priority?

**Comment by darshankawar** [comment]
The root cause seems to be mentioned here: https://github.com/firebase/flutterfire/issues/9117#issuecomment-1187305487
A fix for this can be followed up here: https://github.com/firebase/flutterfire/pull/9138

**Comment by fballan93** [comment]
My web app broke after updating to this latest version . im glade to see this reported . any ETA for the fix

**Comment by mrleesin** [comment]
Same issue

**Comment by woutervanwijk** [comment]
Same here. I'm really amazed how bad QU apparently is that this bug is not catched in the process. It's a big one

**Comment by sgehrman** [comment]
Another good example of the more tests a code base has, the worse the quality. Coders spent too much time writing tests, and not enough time writing quality code. And they use the tests as an excuse for sloppiness as long as their tests pass. This bug is obvious. Should never have been released.

**Comment by mklepaczko** [comment]
Same here for web

**Comment by woutervanwijk** [comment]
> Another good example of the more tests a code base has, the worse the quality. Coders spent too much time writing tests, and not enough time writing quality code. And they use the tests as an excuse for sloppiness as long as their tests pass. This bug is obvious. Should never have been released.

Fully agree, it's a mess. And it's not the first time

**Comment by cedvdb** [comment]
> Another good example of the more tests a code base has, the worse the quality. Coders spent too much time writing tests, and not enough time writing quality code. And they use the tests as an excuse for sloppiness as long as their tests pass. This bug is obvious. Should never have been released.

I cannot disagree more. 
  - The tests are just written in a way that renders them useless.
  - Last time I checked, there was no integration test for phone authentication on the web
 

So the issue is: there is no test.

**Comment by organicinternet** [comment]
#9138 persists for me using these dependency_overrides:

```
  firebase_auth:
    git:
      url: https://github.com/firebase/flutterfire.git
      path: packages/firebase_auth/firebase_auth
      ref: "@russell/auth-9117"
  firebase_auth_platform_interface:
    git:
      url: https://github.com/firebase/flutterfire.git
      path: packages/firebase_auth/firebase_auth_platform_interface
      ref: "@russell/auth-9117"
  firebase_auth_web:
    git:
      url: https://github.com/firebase/flutterfire.git
      path: packages/firebase_auth/firebase_auth_web
      ref: "@russell/auth-9117"
  firebase_core_platform_interface:
    git:
      url: https://github.com/firebase/flutterfire.git
      path: packages/firebase_core/firebase_core_platform_interface
      ref: "@russell/auth-9117"

```

**Comment by kaumudpa** [comment]
Same issue with `firebase_auth 3.5.1`!

**Comment by cosmicb0y** [comment]
Same issue

**Comment by bigbrain-cmd** [comment]
same issue, production is down because of it...

**Comment by stx** [comment]
Firebase is an amazing product, and Flutterfire is an amazing integration, and I think the bug is totally understandable - software will be software - but it feels very scary to me that this issue has been open for 2 weeks in a stable channel, and it received a stable production release anyway, and seemingly has received no response or fix (that I've seen, anyway - I could be wrong.)

I think teams (including ours) are looking for assurance of some kind. I realize this isn't fair to the maintainers, just trying to express how I think other development teams may feel. There is an incredible reliance on Firebase, and the Auth product in particular, in the Flutter industry, and it feels scary to have to deal with these issues on our own.

**Comment by cedvdb** [comment]
> I think the bug is totally understandable - software will be software

That's why integration tests exists.

**Comment by LiangHuangBC** [comment]
> Firebase is an amazing product, and Flutterfire is an amazing integration, and I think the bug is totally understandable - software will be software - but it feels very scary to me that this issue has been open for 2 weeks in a stable channel, and it received a stable production release anyway, and seemingly has received no response or fix (that I've seen, anyway - I could be wrong.)
> 
> I think teams (including ours) are looking for assurance of some kind. I realize this isn't fair to the maintainers, just trying to express how I think other development teams may feel. There is an incredible reliance on Firebase, and the Auth product in particular, in the Flutter industry, and it feels scary to have to deal with these issues on our own.

Agree, similar issue #9166  looks nobody really did test on all platform release.

**Comment by kekko7072** [comment]
+ 1

**Comment by KristofVDB1** [comment]
+ 1

**Comment by sgehrman** [comment]
I've literally been sitting here idle patiently waiting for two weeks now.  Is this fixed yet?  What is going on?  I'm running a business and need to update my software. Revert back to the old build if you have too.  I tried to revert back and couldn't get it working. Could someone just fix it and post a new build?

**Comment by darshankawar** [comment]
The team is actively working on fixing this https://github.com/firebase/flutterfire/pull/9138. Please subscribe to the PR for updates.

**Comment by russellwheatley** [comment]
This fix has just been released in the latest version `3.6.0`. Thanks.

**Comment by cruftmonger** [comment]
I upgraded to 3.6.0 but see the same problem. I updated my original issue with a new dependency list. The rest of the problem description remains the same. https://github.com/firebase/flutterfire/issues/9241#issue-1319713520

**Comment by kaumudpa** [comment]
True. 

There is another problem with the auth state in web since firebase_auth: 3.5.0 - If browser page is refresh/reloaded after the initial authentication of a user sends out unauthorised while trying to access firestore.

**Comment by kaumudpa** [comment]
Also check #9243

**Comment by cruftmonger** [comment]
> This fix has just been released in the latest version `3.6.0`. Thanks.

I still get the same problem with 3.6.0. Did you add a regression test for it? https://github.com/firebase/flutterfire/issues/9241

**Comment by AdamLBS** [comment]
On my side it has been fixed while upgrading to ``3.6.0`` whenever the page is refreshed I'm still logged in, current dependencies are : 

``firebase_core: ^1.20.0
  firebase_auth: ^3.6.0
  cloud_firestore: ^3.4.1``

**Comment by RajendraAVerma** [comment]
After waiting for about 2 week, issue has been fixed. thanks ❤️.
